### PR TITLE
Updated testing guidelines

### LINKED ISSUE #118: Improve Instructions on Testing
Based on #112 , the testing methodology is out of date in `contributing.md`. This needs to be updated!

### LINKED ISSUE #129: Improve Development Guidelines
**Is your feature request related to a problem? Please explain.**
Confusing to keep track of PR branches. 

**Describe the solution you'd like**
Revamp PR syntax to something like `tcp-documentation-[descriptive-name]`. Update `Contributing.md` guidelines.

### PR DESCRIPTION
**Pull request by TheCedarPrince**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [x] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I update the `TODO.md` (if applicable)?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [x] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [x] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [x] Did I properly add test dependencies to the `test` directory (if applicable)?
- [x] Did I check relevant tutorials that may be affected by changes in this PR?
- [x] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #129 
Closes #118 

**How did you address these issues with this PR? What methods did you use?**

I added in a formatting toml and updating the testing guidelines.

### CODE REVIEW

File: docs/src/contributing.md [hunk 1]
```diff
@@ -99,3 +119,43 @@ Finally, click "Create pull request".
 You may get some questions about it, and possibly suggestions of how to make it ready to go into the main project. 
 If you had test errors or problems, we are happy to help you. 
 Then, if all goes according to plan, it gets merged... **Thanks for the contribution!!** :tada: :tada: :tada:
+
+## Note on Adding Dependencies
+
```
Comment (Wikunia) [inline docs/src/contributing.md:124]: Maybe mention that we try to not have too many dependencies? 
I can't remember if it's mentioned somewhere so if not: The developer should create an issue first before working on a PR.

**Review by Wikunia — CHANGES_REQUESTED** [review]
Thanks looks good. Just two things that might be nice to have

File: docs/src/contributing.md [hunk 2]
```diff
@@ -122,6 +122,9 @@ Then, if all goes according to plan, it gets merged... **Thanks for the contribu
 
 ## Note on Adding Dependencies
 
+As a rule, we try to avoid having too many dependencies.
+Therefore, we request that if you have a PR that adds a new dependency, please have opened an issue previously and gotten express permission from `Javis.jl` maintainers to work on this.
```
Comment (Wikunia) [inline docs/src/contributing.md:126]: I think they should open an issue even if they don't add a dependency. Otherwise someone else might works on it. They can open the issue and work on the change directly. No permission needed for dependencies it would probably be good if we can say them beforehand that we don't want this. I would rephrase the word "permission" a bit :smile:

Comment (TheCedarPrince) [reply docs/src/contributing.md:126]: Reworked this!

**Review by Wikunia — APPROVED** [review]
